src/coreclr/jit/assertionprop.cpp

```diff
@@ -6157,6 +6157,7 @@ Compiler::fgWalkResult Compiler::optVNConstantPropCurStmt(BasicBlock* block, Sta
         case GT_ARR_LENGTH:
             break;
 
+        case GT_OBJ:
```

Comment:
```suggestion
        case GT_OBJ:
        case GT_BLK:
```
It is in general a good idea to treat `BLK` and `OBJ` equivalently. One of them is probably going to go at some point.